system-addon/lib/HighlightsFeed.jsm

```diff
@@ -112,14 +111,20 @@ this.HighlightsFeed = class HighlightsFeed {
         this.fetchImage(page);
       }
 
+      // When we bookmark something it adds a history entry - so adjust the type
```

Comment:
This comment is a bit misleading. The reason why the type switches to "bookmark" is that a highlight that comes in as history/visited might happen to be bookmarked, so show it as bookmarked even though the source was from history.

Reply:
These were my steps to reproduce for this that caused me to write this comment (on a clean profile):
1. Bookmark a trending pocket story from the context menu
2. Open debugger and set a breakpoint in HighlightsFeed.jsm right after we dedupe against top sites
3. Force a refresh in Higlights (I dismissed another trending story)
4. Observe ```this.store.getState().TopSites.rows``` and see that the first item in the array is the bookmarked trending story with type 'history' ... even though I've never actually visited it.

So it's a bit confusing that it's there even though I never actually visited the page. How about "Adjust the type for 'history' items that are also 'bookmarked'" for the comment?

Reply:
https://github.com/mozilla/activity-stream/pull/3734